src/core_modules/capture-core/components/WidgetsChangelog/common/hooks/useChangelogData.js

```diff
@@ -67,54 +90,103 @@ export const useChangelogData = ({
         },
     );
 
-    const records: ?Array<ChangelogRecord> = useMemo(() => {
-        if (!data) return undefined;
-
-        return data.changeLogs.map((changelog) => {
-            const { change: apiChange, createdAt, createdBy } = changelog;
-            const elementKey = Object.keys(apiChange)[0];
-            const change = apiChange[elementKey];
-
-            const { metadataElement, fieldId } = getMetadataItemDefinition(
-                elementKey,
-                change,
-                dataItemDefinitions,
-            );
-
-            if (!metadataElement) {
-                log.error(errorCreator('Could not find metadata for element')({
-                    ...changelog,
-                }));
-                return null;
-            }
-
-            const { firstName, surname, username } = createdBy;
-            const { options } = metadataElement;
+    useEffect(() => {
+        const fetchRecords = async () => {
+            if (!data) return;
 
-            const previousValue = convert(
-                convertServerToClient(change.previousValue, metadataElement.type),
-                metadataElement.type,
-                options,
+            const mostRecentCreatedAt = data.changeLogs.reduce(
+                (latest, record) => (moment(record.createdAt).isAfter(latest) ? record.createdAt : latest),
+                data.changeLogs[0]?.createdAt,
             );
 
-            const currentValue = convert(
-                convertServerToClient(change.currentValue, metadataElement.type),
-                metadataElement.type,
-                options,
+            const fetchedRecords = await Promise.all(
+                data.changeLogs.map(async (changelog) => {
+                    const { change: apiChange, createdAt, createdBy, type } = changelog;
+                    const elementKey = Object.keys(apiChange)[0];
+                    const change = apiChange[elementKey];
+
+                    const { metadataElement, fieldId } = getMetadataItemDefinition(
+                        elementKey,
+                        change,
+                        dataItemDefinitions,
+                    );
+
+                    if (!metadataElement) {
+                        log.error(
+                            errorCreator('Could not find metadata for element')({ ...changelog }),
+                        );
+                        return null;
+                    }
+
+                    const getSubValue =
+                subValueGetterByElementType[RECORD_TYPE[entityType]]?.[metadataElement.type];
+
+                    const isLatestValue = moment(createdAt).isSameOrAfter(mostRecentCreatedAt);
```

Comment:
When there is more than one FILE or IMAGE in the form, the URL/preview is shown for only one of them. I think this error is related to how `isLatestValue` is calculated. Can you take a look?

https://github.com/user-attachments/assets/2db89b1c-d705-4a0d-88c7-335d0d07151a

Reply:
Thanks. The code is now updated to find the latest value for each field with **RECORD_TYPE** `trackedEntity` or `event`